plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -0,0 +1,303 @@
+        property of
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset`).
+
+    func: sub-class of `~plasmapy.analysis.fit_functions.AbstractFitFunction`
+        The callable function :math:`f(x)` repressing the fit and its results.
+
+    islands: `List[slice]`
+        List of `slice` objects representing the indices of the identified
+        crossing-islands.
+
+    indices: `slice`
+        A `slice` object representing the indices of ``voltage`` and ``current``
+        arrays used for the fit.
+
+    """
+    rtn = FloatingPotentialResults(
+        vf=np.nan,
+        vf_err=np.nan,
+        rsq=None,
+        func=None,
+        islands=None,
+        indices=None
+    )._asdict()
+
+    fit_funcs = {
+        "linear": {
+            "func": ffuncs.Linear(),
+            "min_point_factor": 0.1,
+        },
+        "exponential": {
+            "func": ffuncs.ExponentialPlusOffset(),
+            "min_point_factor": 0.2,
+        },
+    }
+    try:
+        min_point_factor = fit_funcs[fit_type]["min_point_factor"]
+        fit_func = fit_funcs[fit_type]["func"]
+        rtn["func"] = fit_func
+    except KeyError:
+        raise KeyError(
+            f"Requested fit function '{fit_type}' is  not a valid option.  "
+            f"Examine kwarg 'fit_curve' for valid options."
+        )
+
+    if current.min() > 0.0 or current.max() < 0:
+        warn("The Langmuir sweep has no floating potential.")
+
+        return FloatingPotentialResults(**rtn)
+
+    # check voltage is monotonically increasing/decreasing
+    voltage_diff = np.diff(voltage)
+    if not (np.all(voltage_diff >= 0) or np.all(voltage_diff <= 0)):
+        warn("The voltage array is not monotonically increasing or decreasing.")
+
+        return FloatingPotentialResults(**rtn)
+
+    # condition kwarg threshold
+    if isinstance(threshold, (int, float)):
+        threshold = int(np.round(threshold))
+        if threshold < 1:
+            warn(f"threshold ({threshold}) is less than 1 and needs to be"
+                 f" an int >= 1, using a value of 1")
+            threshold = 1
+    else:
+        warn(f"threshold is NOT a integer >= 1, using a value of 1")
+
+    # condition min_points
+    if min_points is None:
+        min_points = int(np.max([5, np.around(min_point_factor * voltage.size)]))
+    elif min_points == 0:
+        # this signals to use all points
+        pass
+    elif 0 < min_points < 1:
+        min_points = int(np.round(min_points * voltage.size))
+    elif min_points >= 1:
+        min_points = int(np.round(min_points))
+    else:
+        raise ValueError(f"Got {min_points}, but 'min_points' must be an int or float "
+                         f"greater than or equal to 0.")
+
+    # find possible crossing points (cp)
+    lower_vals = np.where(current <= 0, True, False)
+    upper_vals = np.where(0 <= current, True, False)
+    cp_exact = np.logical_and(lower_vals, upper_vals).nonzero()[0]
```

Comment:
```suggestion
    cp_exact = (lower_vals & upper_vals).nonzero()[0]
```
0, because...?

Reply:
Note to self: quitting here because the warnings are becoming too dense. Test it out, please!

Reply:
There does not seem to be a significant difference in performance between the two options.  I feel `np.logical_and` is more explicit, so I'm going to stick with that.

<img width="599" alt="image" src="https://user-images.githubusercontent.com/29869348/94307904-7da69580-ff2a-11ea-82f6-0b3eb19905ec.png">